src/Symfony/Component/Routing/Loader/XmlFileLoader.php

```diff
@@ -115,8 +115,9 @@ protected function parseRoute(RouteCollection $collection, \DOMElement $node, $p
         $methods = preg_split('/[\s,\|]++/', $node->getAttribute('methods'), -1, PREG_SPLIT_NO_EMPTY);
 
         list($defaults, $requirements, $options, $condition) = $this->parseConfigs($node, $path);
+        $path = $node->getAttribute('path');
 
-        $route = new Route($node->getAttribute('path'), $defaults, $requirements, $options, $node->getAttribute('host'), $schemes, $methods, $condition);
+        $route = new Route('null' !== $path ? $path : null, $defaults, $requirements, $options, $node->getAttribute('host'), $schemes, $methods, $condition);
```

Comment:
this is not a real null value in xml I assume. you need to make the path in the xsd `nillable="true"`: https://github.com/nicolas-grekas/symfony/blob/392747ceed64a61ffef636e05f837a13e169e8cc/src/Symfony/Component/Routing/Loader/schema/routing/routing-1.0.xsd#L40

Reply:
actually that's not possible on attributes, so we have to do as implemented I suppose, deal with `null` as a special case

Reply:
Yeah I also figured. But I think it would be better to make path attribute optional in xml instead. null as a string is a hack and potential bc break.

Reply:
> make path attribute optional in xml instead

I fear this will hurt DX, as people will more easilly forget to add the path without having any reminder

> null as a string is a hack and potential bc break

let's assume nobody configured a route on path "null", and that those who did wrote it as "/null", which is what we all do, isn't it?

Reply:
> I fear this will hurt DX, as people will more easilly forget to add the path without having any reminder

Good point. Another brainstorming idea: Add a new attribute like `empty-path=true|false` that defaults to false for BC. So one could use `path="" empty-path=true` to achive that. Does not look nice but path="null" is not transform meaning at all and if you know xml (likely whhen you use xml. config), you'd know it's not a real null value.

Reply:
brainstorming: is it possible to use ~ also in xml like: <route name="abc" path="~" /> ?

Reply:
Done using `empty-path="true"`